subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r76/SystemPropertyPropagationCrossVersionTest.groovy

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.tooling.r76
+
+import org.gradle.integtests.tooling.fixture.TargetGradleVersion
+import org.gradle.integtests.tooling.fixture.ToolingApiSpecification
+import org.gradle.integtests.tooling.fixture.ToolingApiVersion
+import org.gradle.tooling.BuildLauncher
+
+@ToolingApiVersion('>=7.6')
+@TargetGradleVersion('>=7.6')
+class SystemPropertyPropagationCrossVersionTest extends ToolingApiSpecification {
+
+    OutputStream out
+
+    def setup() {
+        toolingApi.requireDaemons()
```

Comment:
Does this require daemons because the embedded executer always sees the system properties?

This could effect real builds since TestKit uses TAPI embedded execution

Reply:
The embedded executor has a different code path that I didn't consider in my change. I'm looking into modifying the embedded executor, but there will be differences due to the missing process separation

Reply:
[`fe8f7cd` (#21223)](https://github.com/gradle/gradle/pull/21223/commits/fe8f7cd86ef79c7390824d38211f7a2a6d956308)